ubuntu-advantage

```diff
@@ -165,11 +177,28 @@ disable_esm() {
 }
 
 fips_prep_check() {
-    local fips_hmacs pkg
+    local fips_hmacs pkg power_cpu_ver
 
     # sanity check
     case "$ARCH" in
-        s390x|x86_64|ppc64le) ;;
+        x86_64)
+            if ! check_cpu_flag aes; then
+                echo 'FIPS requires AES CPU extensions'
+                return 1
+            fi
+            ;;
+
+        ppc64le)
+            power_cpu_ver="$(power_cpu_version)"
+            if [ -z "$power_cpu_ver" ] || [ "$power_cpu_ver" -lt 8 ]; then
+                echo 'FIPS not supported on this CPU'
```

Comment:
I think we could say here that we require POWER8 or higher. Something like this perhaps: "Unsupported POWER cpu: 8 or higher is required", or "FIPS requires POWER version 8 or higher", or "FIPS requires POWER8 or later". Better ask the power people how this "version" is usually referred to.

Reply:
https://en.wikipedia.org/wiki/POWER8
Non Authorative answer, but "FIPS requires POWER8 or later" sounds good to me.